packages/shared/lib/services/plans/plans.ts

```diff
@@ -0,0 +1,71 @@
+import ms from 'ms';
+
+import { Err, Ok } from '@nangohq/utils';
+
+import type { DBPlan } from '@nangohq/types';
+import type { Result } from '@nangohq/utils';
+import type { Knex } from 'knex';
+
+export const TRIAL_DURATION = ms('15days');
+
+export async function getPlan(db: Knex, { accountId }: { accountId: number }): Promise<Result<DBPlan>> {
+    try {
+        const res = await db.from<DBPlan>('plans').select<DBPlan>('*').where('account_id', accountId).first();
+        return res ? Ok(res) : Err(new Error('unknown_plan_for_account'));
+    } catch (err) {
+        return Err(new Error('failed_to_get_plan', { cause: err }));
+    }
+}
+
+export async function createPlan(
+    db: Knex,
+    { account_id, ...rest }: Pick<DBPlan, 'account_id' | 'name'> & Partial<Omit<DBPlan, 'account_id' | 'created_at' | 'updated_at'>>
+): Promise<Result<DBPlan>> {
+    try {
+        const res = await db
+            .from<DBPlan>('plans')
+            .insert({
+                trial_start_at: new Date(),
+                trial_end_at: new Date(Date.now() + TRIAL_DURATION),
+                ...rest,
+                created_at: new Date(),
+                updated_at: new Date(),
+                account_id
+            })
+            .onConflict('account_id')
+            .ignore()
+            .returning('*');
+        return Ok(res[0]!);
```

Comment:
you can do `const [res] = ...` above and have a more pleasing to the eye `return(Ok(res))` 😁